deltacat/compute/compactor/model/round_completion_info.py

```diff
@@ -31,7 +47,7 @@ def of(
         return rci
 
     @property
-    def high_watermark(self) -> Union[Dict[PartitionLocator, int], int]:
+    def high_watermark(self) -> Union[HighWaterMark, int]:
```

Comment:
The return doesn't need to be `Union` then.

Reply:
I assume other users can still a simple integer as the high watermark